tests/test_templates_embeddings.py

```diff
@@ -328,6 +330,117 @@ def circuit(x=None):
             circuit(x=[2, 3])
 
 
+class TestQAOAEmbedding:
+    """ Tests the QAOAEmbedding method."""
+
+    # TODO: Do we need to test all possible local fields?
+
+    def test_qaoa_embedding_state_zero_weights(self, qubit_device, n_subsystems):
+        """Checks the state produced by QAOAEmbedding() is correct if the weights are zero."""
+
+        features = [pi, pi/2, pi/4, 0]
+        if n_subsystems == 1:
+            shp = (1, 1)
+        elif n_subsystems == 2:
+            shp = (1, 3)
+        else:
+            shp = (1, 2 * n_subsystems)
+
+        weights = np.zeros(shape=shp)
+
+        @qml.qnode(qubit_device)
+        def circuit(x=None):
+            QAOAEmbedding(features=x, weights=weights, wires=range(n_subsystems))
+            return [qml.expval(qml.PauliZ(i)) for i in range(n_subsystems)]
+
+        res = circuit(x=features[:n_subsystems])
+        target = [1, -1, 0, 1, 1]
+        assert np.allclose(res, target[:n_subsystems])
```

Comment:
should the assert statements have tolerance set, e.g., 
```suggestion
        assert np.allclose(res, target[:n_subsystems], atol=tol, rtol=0)
```